### PR TITLE
JDK-8292067 Convert test/sun/management/jmxremote/bootstrap shell tests to java version

### PR DESCRIPTION
**Pull request by bwhuang-us**
This task convert 3 shell tests below to java version. 
test/sun/management/jmxremote/bootstrap/RmiBootstrapTest.sh test/sun/management/jmxremote/bootstrap/RmiSslBootstrapTest.sh 
test/sun/management/jmxremote/bootstrap/RmiSslNoKeyStoreTest.sh

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292067](https://bugs.openjdk.org/browse/JDK-8292067): Convert test/sun/management/jmxremote/bootstrap shell tests to java version

### Reviewers
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9973/head:pull/9973` \
`$ git checkout pull/9973`

Update a local copy of the PR: \
`$ git checkout pull/9973` \
`$ git pull https://git.openjdk.org/jdk pull/9973/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9973`

View PR using the GUI difftool: \
`$ git pr show -t 9973`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9973.diff">https://git.openjdk.org/jdk/pull/9973.diff</a>

</details>

### CODE REVIEW

**Comment by bwhuang-us** [comment]
@lmesnik Could you please review this PR?

File: test/jdk/sun/management/jmxremote/bootstrap/RmiTestBase.java [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+                s.lastIndexOf("."));
+
+        List<Path> propertyFiles = Utils.copyFiles(files, Paths.get(DEST),
+                removeSuffix,
+                StandardCopyOption.REPLACE_EXISTING);
+
+        if (isWindows) {
+            Utils.replaceFilesString(propertyFiles,
+                    (s) -> s.replace(TEST_SRC, DEST)
+                            .replaceAll("[/\\\\]", "\\\\\\\\"));
```
Comment (lmesnik) [inline test/jdk/sun/management/jmxremote/bootstrap/RmiTestBase.java:162]: Could you please comment this replacement.

File: test/jdk/sun/management/jmxremote/bootstrap/RmiBootstrapTest.java [hunk 2]
```diff
@@ -20,22 +20,17 @@
...
 import java.rmi.server.ExportException;
 
-import java.util.Properties;
-import java.util.Iterator;
-import java.util.Set;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Enumeration;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (bwhuang-us) [reply test/jdk/sun/management/jmxremote/bootstrap/RmiBootstrapTest.java:33]: Good catch! I didn't introduce this intentionally. It was automatically rearranged by IntelliJ. I will find out a way to disable that feature.

File: test/jdk/sun/management/jmxremote/bootstrap/RmiBootstrapTest.java [hunk 3]
```diff
@@ -68,160 +85,70 @@
...
-                                  ObjectName pattern, QueryExp query)
-        throws IOException {
+            ObjectName pattern, QueryExp query)
+            throws IOException {
 
-        final Set names = server.queryNames(pattern,query);
-        for (final Iterator i=names.iterator(); i.hasNext(); ) {
-            ObjectName name = (ObjectName)i.next();
-            log.trace("listMBeans","Got MBean: "+name);
+        final Set names = server.queryNames(pattern, query);
```
[the remark this answers is not in the document]
Comment (bwhuang-us) [reply test/jdk/sun/management/jmxremote/bootstrap/RmiBootstrapTest.java:115]: Actually, I didn't make changes to these lines. They were done by the IDEA auto formatter. But I can simplify it and add template parameters.

File: test/jdk/sun/management/jmxremote/bootstrap/RmiBootstrapTest.java [hunk 4]
```diff
@@ -230,34 +157,40 @@ public static int listMBeans(MBeanServerConnection server,
...
-            new StringBuffer(System.getProperty("java.home")).
-            append(fileSeparator).append("conf").append(fileSeparator).
-            append("management").append(fileSeparator).
-            append(basename);
-        return defaultFileName.toString();
-    }
+    public static void main(String args[]) throws Exception {
+        if (args.length == 0) {
+            throw new IllegalArgumentException("Argument is required for this" +
+                    " test");
```
[the remark this answers is not in the document]
Comment (bwhuang-us) [reply test/jdk/sun/management/jmxremote/bootstrap/RmiBootstrapTest.java:166]: The IDEA splits the lines for a line limit of 80 characters.

**Review by lmesnik — APPROVED** [review]

**Review by lmesnik — APPROVED** [review]

**Pull request closed without merging**